app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -71,10 +78,15 @@ protected void onCreate(Bundle savedInstanceState) {
         downloadProgress = (ProgressBar) findViewById(R.id.progress);
         downloadProgress.setVisibility(View.VISIBLE);
         downloadProgress.setIndeterminate(true);
-        DataDownload download = new DataDownload();
-        download.downloadVersions();
-
         this.findViewById(android.R.id.content).setBackgroundColor(Color.LTGRAY);
+        ConnectivityManager connMgr = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
```

Comment:
Why is this here? Shouldn't the processor be firing the `ShowNetworkDialogEvent`? Maybe the connection is weaker, maybe it takes forever to download the info. Or maybe there is a network timeout. You are just handling one situation here.